backend/src/v2/driver/driver.go

```diff
@@ -436,14 +438,42 @@ func DAG(ctx context.Context, opts Options, mlmd *metadata.Client) (execution *E
 		return execution, fmt.Errorf("ArtifactIterator is not implemented")
 	}
 	isIterator := opts.Task.GetParameterIterator() != nil && opts.IterationIndex < 0
+	// Fan out iterations
 	if execution.WillTrigger() && isIterator {
 		iterator := opts.Task.GetParameterIterator()
-		value, ok := executorInput.GetInputs().GetParameterValues()[iterator.GetItems().GetInputParameter()]
 		report := func(err error) error {
 			return fmt.Errorf("iterating on item input %q failed: %w", iterator.GetItemInput(), err)
 		}
-		if !ok {
-			return execution, report(fmt.Errorf("cannot find input parameter"))
+		// Check the items type of parameterIterator:
+		// It can be "inputParameter" or "Raw"
+		value := structpb.NewNullValue()
```

Comment:
I think it is necessary. The variable needs to be declared outside the switch/case statement, then it can be assigned value inside the switch/case statements. If the variables were declared inside, it would not be accessible outside the statement. (Ref: https://stackoverflow.com/questions/21481288/variable-not-declared-inside-a-if-else-statement)

Reply:
> The variable needs to be declared outside the switch/case statement, then it can be assigned value inside the switch/case statements.

You can still declare it inside a switch case/branch if it's not reused across multiple branches and not used outside the switch scope--didn't check if that's the case.
That said, my comments was more on the initialization part than the declaration part.

Reply:
It is being used outside the switch scope, so I guess it needs to be declared outside as well. Although I don't have to assign it a null value - I can just declare its type, I don't find a method in `structpb` that just declares a `Variable` type. It looks like they need to initialize some value inside the variable.

Reply:
It turned out that I can use `var value *structpb.Value` for the declaration.